task_02/xieyu_wang/graphic.cpp

```diff
@@ -0,0 +1,114 @@
+#include <array>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+constexpr int MAX_ELEMENT = 18;
+struct Data {
+  int from;
+  int edge;
+  int to;
+};
+class VecToString {
+public:
+  std::string getVectors() { return vectors; }
+  std::string getEdges() { return edges; }
+  void setVectors(std::array<Data, MAX_ELEMENT> data) {
+    vectors = "";
+    int maxIndex = getMaxVecIndex(data);
+    for (int i = 0; i <= maxIndex; i++) {
+      vectors += "\t\t{\"id\":" + std::to_string(i) + ",";
+      vectors += "\"edge_ids\": [";
+      for (const auto it : data) {
+        if (it.from == i || it.to == i) {
+          vectors += std::to_string(it.edge) + ",";
+        }
+      }
+      vectors.pop_back();
+      vectors += "]},\n";
+    }
+    vectors.pop_back();
+    vectors.pop_back();
+    vectors += "\n";
+  }
+  void setEdges(std::array<Data, MAX_ELEMENT> data) {
+    edges = "";
+    for (const auto it : data) {
+      edges += "\t\t{\"id\": " + std::to_string(it.edge);
+      edges += ",\"vertex_ids\": [" + std::to_string(it.from) + "," +
+               std::to_string(it.to) + "]},\n";
+    }
+    edges.pop_back();
+    edges.pop_back();
+    edges += "\n";
+  }
+  int getMaxVecIndex(std::array<Data, MAX_ELEMENT> data) {
+    int maxIndex = 0;
+    for (const auto it : data) {
+      if (it.from > maxIndex)
+        maxIndex = it.from;
+      if (it.to > maxIndex)
+        maxIndex = it.to;
+    }
+    return maxIndex;
+  }
+
+private:
+  std::string vectors;
+  std::string edges;
+};
+class WriteTOFile {
+public:
+  void openFile() {
+    try {
+      graph.open("Graphic.json", std::ios::out);
+      if (!graph.is_open()) {
+        throw "can not open file";
+      }
+    } catch (std::string e) {
+      std::cerr << e << std::endl;
+      std::exit(1);
+    }
+  }
+  void write(std::string vectors, std::string edges) {
+    openFile();
+    graph << "{\n\t\"vertices\": [" << std::endl;
+    graph << vectors;
+    graph << "\t],\n";
+    graph << "\t\"edges\": [" << std::endl;
+    graph << edges;
+    graph << "\t]\n}\n";
+    graph.close();
+  }
+
+private:
+  std::ofstream graph;
+};
+int main() {
+  constexpr std::array<Data, MAX_ELEMENT> netWork = {{{0, 0, 1},
```

Comment:
The difference between `constexp` and `const` is what 
- `constexpr` let's compiler to precalculate the value and replace variable with this value. 
- `const` tells compiler that this variable should not be changed while runtime

So in most cases it makes sense to use `const` in local scope and use `constexpr` to define global variables.
In your case just `const` will be enough
```suggestion
  const std::array<Data, MAX_ELEMENT> netWork = {{{0, 0, 1},
```